AppCenterAuth/AppCenterAuth/Internals/Models/MSAADAuthority.m

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import "MSAADAuthority.h"
+
+@implementation MSAADAuthority
+
+static NSString *const kMSTypeKey = @"type";
+
+static NSString *const kMSAudienceKey = @"audience";
+
+static NSString *const kMSTenantIdKey = @"tenant_id";
+
+static NSString *const kMSAuthorityCommonUrl = @"https://login.microsoftonline.com/";
+
+static NSString *const kMSSingleTenantAudience = @"AzureADMyOrg";
+
+static NSString *const kMSMultiTenantAudience = @"AzureADMultipleOrgs";
+
+static NSString *const kMSCommonAudience = @"AzureADandPersonalMicrosoftAccount";
+
+static NSString *const kMSAuthorityTypeAAD = @"AAD";
+
+static NSString *const kMSCommon = @"common";
+
+static NSString *const kMSorganizations = @"organizations";
+
+- (instancetype)initWithDictionary:(NSDictionary *)dictionary {
+  if (!dictionary) {
+    return nil;
+  }
+  if ((self = [super initWithDictionary:dictionary])) {
+    if (dictionary[kMSAudienceKey]) {
+      if ([(NSDictionary *)dictionary[kMSAudienceKey] isKindOfClass:[NSDictionary class]]) {
+        NSDictionary *audience = (NSDictionary *)dictionary[kMSAudienceKey];
+        NSString *tenantId = (NSString * _Nonnull)audience[kMSTenantIdKey];
```

Comment:
Do we need to type-check the contents for these fields or do we just take whatever is in the dictionary and fail later?

Reply:
In android we are trusting the contents and passing along. I guess we can do that here. MSAL will take careof contents.